js/src/main/scala/com/raquo/domtypes/jsdom/defs/package.scala

```diff
@@ -7,18 +7,28 @@ import scala.scalajs.js
 
 package object defs {
 
-  /** Type refinement for events that are fired on input elements.
-    * You can safely use this type for `onChange` and `onSelect` events,
-    * and SOME `onInput` events. See README for more details. */
   @js.native
-  trait InputElementTargetEvent extends dom.Event {
-    override def target: dom.html.Input = js.native
+  //TODO: maybe even ElementTargetEvent[T <: dom.html.Element]?
```

Comment:
We could also be more specific, as we know that e.g. `onChange` has target `dom.Input | dom.Textarea' and `onSearch` has `dom.Input`.

Reply:
Unfortunately `|` is pretty useless in this case. Even though both`dom.Input` and `dom.Textarea` have a `value` property, you can't just access `target.value` if `target: dom.Input | dom.Textarea`.